CMakeLists.txt

```diff
@@ -43,281 +29,65 @@ else()
     set(PROJECT_DLLVERSION)
     set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_INSTALL_PREFIX}/lib/cmake/ /usr/lib/cmake)
 endif()
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
 
 option(USE_SSE    "Build tiny-cnn with SSE library support" ON)
 option(USE_AVX    "Build tiny-cnn with AVX library support" ON)
 option(USE_TBB    "Build tiny-cnn with TBB library support" OFF)
 option(USE_OMP    "Build tiny-cnn with OMP library support" OFF)
 option(USE_NNPACK "Build tiny-cnn with NNPACK library support" OFF)
+option(USE_OPENCV "Build tiny-cnn with OpenCN library support" OFF)
 
 option(BUILD_TESTS "Set to ON to build tests" OFF)
-option(BUILD_EXAMPLES "Set to ON to build examples" ON)
+option(BUILD_EXAMPLES "Set to ON to build examples" OFF)
 option(BUILD_DOCS "Set to ON to build documentation" OFF)
 
-# ----------------------------------------------------------------------------
-#   Find Dependencies
-# ----------------------------------------------------------------------------
+# ---[ Install options
 
-# Using cmake scripts and modules
-list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
-
-if(BUILD_EXAMPLES)
-    find_package(OpenCV)
-    if(OpenCV_FOUND)
-      add_definitions(-DUSE_OPENCV)
-      include_directories(${OpenCV_INCLUDE_DIRS})
-      set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${OpenCV_LIBS})
-    endif()
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install" 
+        CACHE PATH "Default install path" FORCE)
 endif()
 
-if(USE_TBB)
-    find_package(TBB REQUIRED)
-    add_definitions(-DCNN_USE_TBB)
-    include_directories(${TBB_INCLUDE_DIRS})
-    link_directories(${TBB_LIBRARY_DIRS})
-    set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${TBB_LIBRARIES})
-elseif(USE_OMP)
-    find_package(OpenMP REQUIRED)
-    add_definitions(-DCNN_USE_OMP)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-endif()
-
-if(USE_NNPACK)
-   find_package(NNPACK REQUIRED)
-   add_definitions(-DCNN_USE_NNPACK)
-   include_directories(SYSTEM ${NNPACK_INCLUDE_DIR})
-   include_directories(SYSTEM ${NNPACK_INCLUDE_DIR}/../third-party/pthreadpool/include)
-   set(REQUIRED_LIBRARIES ${REQUIRED_LIBRARIES} ${NNPACK_LIB})
-endif()
+# Using cmake scripts and modules
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
 
-# ----------------------------------------------------------------------------
-# COMPILER OPTIONS
-# ----------------------------------------------------------------------------
+include(cmake/summary.cmake)
+include(cmake/config_gen.cmake)
```

Comment:
I don't think we should put these in separate files. They're rather short (15-40 lines) and in my opinion it's much easier to get an overview of the CMake build and reason about it if there is no "magic" happening in external files and you have to look them up first to see what's going on.